app/backend/src/couchers/utils.py

```diff
@@ -59,3 +63,47 @@ def get_coordinates(geom):
     shp = to_shape(geom)
     # note the funiness with 4326 normally being (x, y) = (lng, lat)
     return (shp.y, shp.x)
+
+
+def http_date(dt=None):
+    """
+    Format the datetime for HTTP cookies
+    """
+    if not dt:
+        dt = now()
+    return formatdate(dt.timestamp(), usegmt=True)
+
+
+def create_session_cookie(token, expiry):
+    cookie = http.cookies.Morsel()
+    cookie.set("couchers-sesh", token, token)
+    # tell the browser when to stop sending the cookie
+    cookie["expires"] = http_date(expiry)
+    # restrict to our domain, note if there's no domain, it won't include subdomains
+    cookie["domain"] = config["COOKIE_DOMAIN"]
```

Comment:
Can't we make one of the http proxies fill in the domain for us? Those are the only ones that knows the real domain.

Reply:
That sounds like a security issue waiting to happen...

Reply:
Not if we set it to some bogus value here, to make sure it is not unset.

Reply:
How do you mean?

Reply:
If we set the cookie domain to "nonexisting.com" here, and then the proxy changes it to "couchers.org", it will work. In case of a misconfiguration where the proxy doesn't change it, the browser will drop the cookie or only send it to "nonexisting.com" => no security issue.

Reply:
That's just pushing the configuration to the proxy?

I meant with the security issue that I'd imagine the idea would be for the proxy to do this somehow dynamically then? That sounds prone to trickery.

Reply:
The proxy already knows which server it is running on (it is serving the https certificate), so no extra configuration needed.

Reply:
Sure, the TLS-terminating nginx knows which domain it lives on.

But really, I think this is app config, and the proxy shouldn't be doing deep header modification just so we can get rid of one line of config in the app? It's much simpler, and also (organisationally) much safer to keep that config and that logic here.

Reply:
Hmm, can't we get the domain from the `Host: ` http header?

Reply:
That's where the security issues start. There's no stopping anyone from spoofing that, though depending on nginx, that might not route to the API server then. It just seems too risky.

Reply:
ok, fair enough! Thanks for talking me through this, resolving thread :)